### PR TITLE
[BREAKING CHANGE] feat: remove notfound.enableRedirect

### PR DESCRIPTION
**Pull request by dead-horse**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

notfound

##### Description of change

remove `notfound.enableRedirect`, always redirect if `notfound.pageUrl` present, ensure has the same behavior in different environments.

### CODE REVIEW

**Comment by dead-horse** [comment]
there's no  additional informations we can present in non-production environments, so keep it the same as production environment is more reasonable.

File: app/middleware/notfound.js [hunk 1]
```diff
@@ -18,16 +18,11 @@ module.exports = options => {
       return;
     }
 
-    if (options.enableRedirect && options.pageUrl) {
+    if (options.pageUrl) {
```
Comment (fengmk2) [inline app/middleware/notfound.js]: 还要防止死循环，确保 referer 不是自己，免得开发者配置了 `options.pageUrl ='/404'`，然后又没有实现 404 页面

Comment (dead-horse) [inline app/middleware/notfound.js]: 比较难，例如 referrer 为 `http://dev.xxx.com/404`，没法判断它是自己，这个还是交由开发者自己保证吧，开发时会发现的。

Comment (fengmk2) [inline app/middleware/notfound.js]: 如果 pageUrl 是当前站点的，以 `/` 开头，可以通过判断 `this.path` 是否等于 pageUrl 来解决的

File: app/middleware/notfound.js [hunk 2]
```diff
@@ -18,16 +18,19 @@ module.exports = options => {
       return;
     }
 
-    if (options.enableRedirect && options.pageUrl) {
+    const htmlTitle = '<h1>404 Not Found</h1>';
+
+    // notfound handler is unimplemented
+    if (options.pageUrl && this.path === options.pageUrl) {
+      this.body = `${htmlTitle}config.notfound.pageUrl(${options.pageUrl}) is unimplemented`;
```
Comment (dead-horse) [inline app/middleware/notfound.js]: 加上了，没有做环境判断的区分，统一返回这个文案了。

File: app/middleware/notfound.js [hunk 3]
```diff
@@ -18,16 +18,19 @@ module.exports = options => {
       return;
     }
 
-    if (options.enableRedirect && options.pageUrl) {
+    const notFoundHtml = '<h1>404 Not Found</h1>';
+
+    // notfound handler is unimplemented
+    if (options.pageUrl && this.path === options.pageUrl) {
+      this.body = `${notFoundHtml}config.notfound.pageUrl(${options.pageUrl}) is unimplemented`;
```
Comment (fengmk2) [inline app/middleware/notfound.js]: 加个 p

``${notFoundHtml}
<p><pre><code>config.notfound.pageUrl(${options.pageUrl})</code></pre> is unimplemented</p>`;`

**Comment by popomore** [comment]
这个配置好像影响不大，就不记录了

**Pull request merged**